bindings/python/rgbmatrix/core.pyx

```diff
@@ -193,45 +274,64 @@ cdef class RGBMatrixOptions:
         def __get__(self): return self.__runtime_options.drop_privileges
         def __set__(self, uint8_t value): self.__runtime_options.drop_privileges = value
 
+    @property
+    def do_gpio_init(self):
+        return self.__runtime_options.do_gpio_init
+    @do_gpio_init.setter
+    def do_gpio_init(self, uint8_t value):
+        self.__runtime_options.do_gpio_init = value
+
 cdef class RGBMatrix(Canvas):
-    def __cinit__(self, int rows = 0, int chains = 0, int parallel = 0,
-        RGBMatrixOptions options = None):
+    def __cinit__(self, rows = None, chains = None, parallel = None,
+                  RGBMatrixOptions options = None):
 
         # If RGBMatrixOptions not provided, create defaults and set any optional
         # parameters supplied
-        if options == None:
+        if options is None:
             options = RGBMatrixOptions()
 
-        if rows > 0:
+        if rows is not None:
             options.rows = rows
-        if chains > 0:
+        if chains is not None:
             options.chain_length = chains
-        if parallel > 0:
+        if parallel is not None:
             options.parallel = parallel
 
         self.__matrix = cppinc.CreateMatrixFromOptions(options.__options,
-            options.__runtime_options)
+                                                       options.__runtime_options)
 
-    def __dealloc__(self):
-        self.__matrix.Clear()
-        del self.__matrix
+        if self.__matrix == NULL:
+            raise Exception("RGBMatrix creation failed")
 
-    cdef cppinc.Canvas* __getCanvas(self) except *:
-        if <void*>self.__matrix != NULL:
-            return self.__matrix
-        raise Exception("Canvas was destroyed or not initialized, you cannot use this object anymore")
+    def __init__(self, *args, **kwargs):          # no call to Canvas.__init__
+        pass
 
-    def Fill(self, uint8_t red, uint8_t green, uint8_t blue):
-        self.__matrix.Fill(red, green, blue)
+    def __dealloc__(self):
+        if self.__matrix != NULL:
+            self.__matrix.Clear()
+            del self.__matrix
 
-    def SetPixel(self, int x, int y, uint8_t red, uint8_t green, uint8_t blue):
-        self.__matrix.SetPixel(x, y, red, green, blue)
+    cdef cppinc.Canvas* __getCanvas(self) except NULL:
+        assert self.__matrix != NULL
```

Comment:
Generally: Are these assertions required? If so, can we at least add an error message?